### DISCUSSION TITLE
What kind of sync is applied to M3Us?

### DISCUSSION THREAD

**Discussion by Pfeil** [navidrome/navidrome#2979] [Q&A]
I read on the [feature overview](https://www.navidrome.org/docs/overview/#features): "Full support for playlists, with option to auto-import .m3u files **and to keep them in sync**"

I tried this, but it seems not to "keep them in sync" (which I read as 2-way-sync) but rather auto-import and auto-update (or 1-way-"sync" from playlist to navidrome). It also overrides changes I may have made on the server (overrides on file update). I wonder which I am holding wrong: navidrome or its manual? Is this intended?

Background/Goal: I sync this folder between devices and would like to have my playlist also in the form of a file in order to use it in other devices. This also means I need actually relative paths in my playlists (which is an issue with probably all media players). 

If this is not a bug, the only workaround I see is to export the playlist manually (can I automate this?) in order to "sync back" and have a cron job to convert paths back to relative. Is there another one?

**Comment by kgarner7** [comment]
Navidrome will import m3u files, and if auto-import is set it will continue to keep the playlist in Navidrome sync. It will never modify your library (e.g., by modifying the file). If you want to have changes from Navidrome reflected in the file, you'll need do do something like `navidrome pls -p $PLAYLIST_ID -o $OUTPUT_M3U_FILE` (may need additional flags).

**Reply by Pfeil** [reply]
Thanks, this was really helpful. In which cases will it auto-import again? For example, is there a time frame in which I need to "save back" my changes from Navidrome to the file before it overrides the list in navidrome? Or will it trigger auto-import only if the file has been modified? What about navidrome reboots and other edge cases?

**Reply by kgarner7** [reply]
Playlist imports/updates happen whenever you scan your library. If you haven't changed the scan schedule, it happens every minute
